pkg/controllers/dynakube/activegate/internal/authtoken/reconciler.go

```diff
@@ -128,7 +131,7 @@ func (r *Reconciler) createSecret(ctx context.Context, secretData map[string][]b
 		return errors.WithStack(err)
 	}
 
-	err = r.client.Create(ctx, secret)
+	err = r.query().WithOwner(r.dk).Create(ctx, secret)
```

Comment:
TBH, i don't get why `r.query()` is better then `r.cleint`? Could you please clarify?

Reply:
It was started out as "refactor it more", but because this `AuthToken` is a bit special (we never update, just delete and recreate), so I couldn't use the nicer parts of the `query`
- only the `Delete` is a tiny bit better
- and now we have an owner-reference 

it is mainly for consistency, we should use our abstraction everywhere, so we handle the secrets in a consistent way
- Example: Have a owner-reference on it
- Maybe future example: Have the hash-annotation on it